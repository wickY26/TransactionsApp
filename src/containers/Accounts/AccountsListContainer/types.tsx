import { ReactElement } from 'react';

import { Account } from '../types';

export interface RenderProps {
  accounts: Account[];
}

export interface AccountsListContainerProps {
  children: (props: RenderProps) => ReactElement;
};
