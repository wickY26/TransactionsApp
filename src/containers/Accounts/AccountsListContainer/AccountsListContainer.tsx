import { FC, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { AccountsListContainerProps } from './types';
import { thunkFetchAccounts } from '../actions';
import { accountsSelector } from '../selectors';

const AccountsListContainer: FC<AccountsListContainerProps> = ({ children }) => {
  const accounts = useSelector(accountsSelector);
  const dispatch = useDispatch();
 
  useEffect(() => {
    dispatch(thunkFetchAccounts());
  }, [dispatch]);

  return children({ accounts });
};

export default AccountsListContainer;
