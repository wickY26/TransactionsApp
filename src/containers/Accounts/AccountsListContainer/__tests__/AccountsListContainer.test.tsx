import React from 'react';
import { render } from '@testing-library/react';

import AccountsListContainer from '..';

const mockAccount = {name:'Poyraz', surname: 'Yilmaz', id: 0};
const mockDispatch = jest.fn();
const mockSelector = jest.fn();
const mockThunkFetchAccount = jest.fn();

jest.mock('react-redux', () => ({
  useSelector: () => mockSelector(),
  useDispatch: () => mockDispatch,
}));

jest.mock('../../actions', () => ({
  thunkFetchAccounts: () => mockThunkFetchAccount(),
}));

const renderComponent = () => render(
  <AccountsListContainer>
    {({accounts}) => (
      <ul>
        {accounts?.map(account => <li key={account.name}>{account.name}</li>)}
      </ul>
    )}
  </AccountsListContainer>
);

describe('AccountsListContainer', () => {

  beforeEach(() => {
    mockSelector.mockReturnValue([mockAccount]);
    mockThunkFetchAccount.mockReturnValue('FetchAccountAction');
  });

  it('should render correctly', () => {
    const component = renderComponent();
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should dispatch fetchAccountsAction', () => {
    renderComponent();
    expect(mockThunkFetchAccount).toHaveBeenCalledTimes(1);
    expect(mockDispatch).toHaveBeenCalledWith('FetchAccountAction');
  });
  
  it('should select accounts from store', () => {
    renderComponent();
    expect(mockSelector).toHaveBeenCalled();
  });

});
