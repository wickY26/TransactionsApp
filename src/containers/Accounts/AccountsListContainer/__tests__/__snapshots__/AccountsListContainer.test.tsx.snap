// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountsListContainer should render correctly 1`] = `
<DocumentFragment>
  <ul>
    <li>
      Poyraz
    </li>
  </ul>
</DocumentFragment>
`;
